src/homepageExperience/components/steps/cli/InstallDependencies.tsx

```diff
@@ -0,0 +1,312 @@
+import CodeSnippet from 'src/shared/components/CodeSnippet'
+import {SafeBlankLink} from 'src/utils/SafeBlankLink'
+import {
+  ComponentColor,
+  Button,
+  ButtonGroup,
+  Orientation,
+  Table,
+  BorderType,
+} from '@influxdata/clockface'
+
+import React, {useEffect, useState} from 'react'
+import {useDispatch} from 'react-redux'
+import {getBuckets} from 'src/buckets/actions/thunks'
+import {event} from 'src/cloud/utils/reporting'
+
+export const InstallDependencies = () => {
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(getBuckets())
+  }, [])
+
+  const logCopyCodeSnippetMac = () => {
+    event('firstMile.cliWizard.installDependenciesMac.code.copied')
+  }
+  const logCopyCodeSnippetWindows = () => {
+    event('firstMile.cliWizard.installDependenciesWindows.code.copied')
+  }
+  const logCopyCodeSnippetLinux = () => {
+    event('firstMile.cliWizard.installDependenciesLinux.code.copied')
+  }
+
+  const windowsCodeSnippet =
+    "> Expand-Archive .\\influxdb2-client-latest-amd64.zip -DestinationPath 'C:\\Program Files\\InfluxData\\'\n" +
+    "> mv 'C:\\Program Files\\InfluxData\\influxdb2-client-latest-windows-amd64' 'C:\\Program Files\\InfluxData\\influx'"
+  const linuxCodeSnippetA = `# amd64
+wget https://dl.influxdata.com/influxdb/releases/influxdb2-client-latest-linux-amd64.tar.gz
+  
+# arm
+wget https://dl.influxdata.com/influxdb/releases/influxdb2-client-latest-linux-arm64.tar.gz
+  `
+  const linuxCodeSnippetB = `# amd64
+tar xvzf path/to/influxdb2-client-latest-linux-amd64.tar.gz
+  
+# arm
+tar xvzf path/to/influxdb2-client-latest-linux-arm64.tar.gz
+  `
+  const linuxCodeSnippetC = `# amd64
+sudo cp influxdb2-client-latest-linux-amd64/influx /usr/local/bin/
+            
+# arm
+sudo cp influxdb2-client-latest-linux-arm64/influx /usr/local/bin/
+  `
+
+  const [macSelected, setMacSelected] = useState(true)
+  const [windowsSelected, setWindowsSelected] = useState(false)
+  const [linuxSelected, setLinuxSelected] = useState(false)
+
+  return (
+    <>
+      <h1>Install Dependencies</h1>
+      <ButtonGroup orientation={Orientation.Horizontal}>
+        <Button
+          text="mac OS"
+          color={macSelected ? ComponentColor.Primary : ComponentColor.Default}
+          onClick={() => {
+            setMacSelected(true)
+            setWindowsSelected(false)
+            setLinuxSelected(false)
```

Comment:
So this wil work, but it's a lot of state to accomplish the goal - there are three different flags that need to be set, and the value of one of them depends on the value of the other two - if `macSelected` is true, then `windowsSelected` and `linuxSelected` must both be false, and it's up to the coder to make sure this happens. These things tend to be hard for most people to keep in their head at once, and they end up being really easy to cause bugs in accidentally.

Rather than having a `isSelected` type flag for each option, let's instead have a single variable that tracks the currently selected item. There are three different options for which OS is selected: Mac, Linux, and Windows. If we called the variable `currentSelection` (or something similar, it doesn't have to be called this), and give it a limited range of values, we could accomplish the same thing with a single variable.

TypeScript has a feature called [string literal types](https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types) that fit this nicely. What we'd do is create a TypeScript type that only allows the narrow range of values we care about, and then set the value of our variable to one of those Typed values.

```ts
type CurrentOSSelection = 'Linux' | 'Mac' | 'Windows'
```
then when using the `useState` hook, we could do something like:

```ts
const [currentSelection, setCurrentSelection] = useState<CurrentOSSelection>('Mac')
```
Then, the `Buttons` become something like:

```ts
color={currentSelection === 'Mac' ? ComponentColor.Primay : ComponentColor.Default}
onClick={() => {
  setCurrentSelection('Mac')
}}
```
and the check for which part to render becomes:
```ts
{currentSelection === 'Mac' && (
...
```